api/v1alpha1/storageclaim_types.go

```diff
@@ -53,7 +53,7 @@ type StorageClientNamespacedName struct {
 
 // StorageClaimSpec defines the desired state of StorageClaim
 type StorageClaimSpec struct {
-	//+kubebuilder:validation:Enum=block;sharedfile
+	//+kubebuilder:validation:XValidation:rule="self.lowerAscii()=='block'||self.lowerAscii()=='sharedfile'",message="value should be either 'sharedfile' or 'block' (case insensitive)"
```

Comment:
```suggestion
//+kubebuilder:validation:Enum=block;sharedFile
```
@leelavg i dont see a reason to change the CEL validation due to the internal details. if there is a problem with `shreadfile`, we can change it to `sharedFile` but we need to have strict validation for the user input.

Reply:
we shouldn't be enforcing the user to choose b/n letters casing, CEL rule allows casing but still enforces the values. I'm not creating rule for internal details, it's providing way for user not to think about what the case should be.

I could mention all the variations in enum as done in ocs for some different field https://github.com/red-hat-storage/ocs-operator/blob/cc58086ab8ed14965e6277c865d6aaa34c8cc49f/api/v1/storagecluster_types.go#L52
 but having CEL rule greatly simplifies the requirement and indicates the intention.

@Madhu-1 ^^

Reply:
> SHaReDFiLE BLOck etc
- I know, should that effect on how we reconcile, is it changing the intent on what we need to do beside the point user is really doing that?

Reply:
It wont but we also need to think about how to keep our CRD as we want, not giving more than expected flexibility to users where CR's looks ugly. i have never seen that kube objects, do you have any example for it?

Reply:
i think you are missing the point, this is not some user-chosen name like CR names, this is the field we are enforcing how it looks like and how it should be, we don't need to give this much flexibility to the user.

IMHO we should keep it as `sharedFile` and `block` nothing else. @nb-ohad thought?

Reply:
As this is a customer-facing CR,  I would like to remove the unnecessary strictness a user needs to consider when filling in the CR. 

A user-friendliness consideration will dictate that all of the values below should be ok:
- sharedfile
- SharedFile
- Sharedfile
- sharedFile

I am not concerned about a user abusing the system to have values like sHaReDfIle.  If someone set the value to be something like that then it was done on purpose, and as a whole, the system is resilient enough to handle such values.

Reply:
@Madhu-1 Please confirm that you are ok with the above so we can merge

Reply:
i have not seen anything like this and if others are agreeing to have this i don't have any objection.